pkg/prometheus/promcfg.go

```diff
@@ -712,6 +724,21 @@ func (cg *ConfigGenerator) generatePodMonitorConfig(
 		cfg = append(cfg, yaml.MapItem{Key: "target_limit", Value: getLimit(m.Spec.TargetLimit, enforcedTargetLimit)})
 	}
 
+	// since label limits are only available from prometheus 2.27
+	if version.Major == 2 && version.Minor >= 27 {
```

Comment:
We should be first checking if `LabelLimit` is set then checking for prometheus version. This way we can emit log messages about unsupported configuration instead of silently ignoring config. Same goes for all config options introduced in this PR.